Swc 187 modificar el endpoint de listar partidas

MatiasScantamburlo opened the pull request:
Las partidas ahora pueden ser privadas
    
    - Cambiamos GameIn agregando el campo is_private, el cual debe de ser completado
      al llamar a método POST /api/lobby.
    - Tambien se cambian GameOut devolviendo al cliente el nuevo campo.
    - Se modifica la DB para persistir este nuevo atributo de la clase Game.
    - Se modifican los test para el cambio.

MatiasScantamburlo committed:
Las partidas ahora pueden ser privadas

- Cambiamos GameIn agregando el campo is_private, el cual debe de ser completado
  al llamar a método POST /api/lobby.
- Tambien se cambian GameOut devolviendo al cliente el nuevo campo.
- Se modifica la DB para persistir este nuevo atributo de la clase Game.
- Se modifican los test para el cambio.
- Se agrean docstrings.

 On branch SWC-187-Modificar-el-endpoint-de-listar-partidas

 Changes to be committed:
	modified:   main.py
	modified:   model/game.py
	modified:   tests/test_create_game.py
	modified:   tests/test_get_game.py

MatiasScantamburlo committed:
format

MatiasScantamburlo committed:
Implementación de contraseñas con hash

- Se agregó el almacenamiento de una contraseña con hash para cada Game
  en la base de datos.
- Se configuró CryptContext como manejador de contraseñas, utilizando
  passlib (v1.7.4) para el hash de contraseñas.
- Se actualizaron las clases *IN y *OUT para permitir el ingreso y
  devolución de la contraseña ingresada.
- En el endpoint POST /api/lobby, se añadieron validaciones de
  contraseña y se aplica el hash a la clave ingresada.
- En el endpoint PUT /api/lobby/{id_game}, se verifica la contraseña
  proporcionada para permitir el ingreso al lobby.

 On branch SWC-187-Modificar-el-endpoint-de-listar-partidas

 Changes to be committed:
 	modified:  main.py
	modified:  model/game.py
	modified:  requirements.txt

KaiserOfNone commented on main.py:
```diff
@@ -81,6 +86,8 @@ class GameOut(BaseModel):
     min_players: int
     started: bool
+    is_private: bool
+    password: Optional[str] = None
```
Nein, no pases el password, aunque sea hasheado es completamente interno

KaiserOfNone commented on main.py:
```diff
@@ -90,23 +97,42 @@ async def create_game(
             status_code=412,
             detail="El número mínimo de jugadores no puede ser mayor al máximo",
         )
+    if game_create.min_players < 2 or game_create.max_players > 4:
+        raise HTTPException(
+            status_code=412,
+            detail="El número de jugadores debe estar entre 2 y 4",
+        )
```
Pydantic enfuerza esto, no hace falta hacerlo a mano.

MatiasScantamburlo replied:
Mile grazie

KaiserOfNone reviewed (changes requested):
Faltan tests. Crea un test que setee un password e intente entrar, que no de el password cuando hace falta etc.

MatiasScantamburlo committed:
logica de verificacion en /ws/lobby/{game_id}

MatiasScantamburlo committed:
Agregamos los test

- Se agregan test en test_notify_lobby para probar la conexion a
  partidas publica, privadas con contraseña correcta y privadas con
  contraseña erronea.

 On branch SWC-187-Modificar-el-endpoint-de-listar-partidas

 Changes to be committed:
	modified:   tests/test_notify_lobby.py

MatiasScantamburlo committed:
Changes requested

MatiasScantamburlo committed:
Merge branch 'develop' into SWC-187-Modificar-el-endpoint-de-listar-partidas

MatiasScantamburlo committed:
format
